oceanstream/cli.py

```diff
@@ -0,0 +1,194 @@
+
+from oceanstream.denoise import (
+    apply_background_noise_removal,
+    apply_noise_masks,
+    apply_seabed_mask,
+    create_masks,
+)
+from oceanstream.echodata import (
+    compute_sv_with_encode_mode,
+    enrich_sv_dataset,
+    read_file,
+    write_processed,
+)
+from oceanstream.exports import compute_and_write_nasc
+from oceanstream.exports import write_csv as write_shoals_csv
+from oceanstream.exports.csv import export_raw_csv, export_Sv_csv
+from oceanstream.exports.plot import plot_all_channels
+from oceanstream.report import display_profiling_and_summary_info
+from oceanstream.settings import load_config
+from oceanstream.utils import attach_mask_to_dataset
+
+DEFAULT_OUTPUT_FOLDER = "output"
+
+
+def parse_cli_arguments():
+    parser = argparse.ArgumentParser(description="Process hydroacoustic data.")
+    parser.add_argument(
+        "--raw-data-source",
+        type=str,
+        required=True,
+        help="Path to a raw data file or directory containing multiple raw data files",
+    )
+    parser.add_argument(
+        "--output-folder",
+        type=str,
+        default=None,
+        help="Destination path for saving processed data. Defaults to a predefined directory if not specified.",
+    )
+    parser.add_argument(
+        "--sonar-model", type=str, help="Sonar model used to collect the data (EK60, EK80, etc.)"
+    )
+    parser.add_argument(
+        "--export-csv", action="store_true", help="Write CSV output files with processed data"
+    )
+    parser.add_argument("--config", type=str, default=None, help="Path to a configuration file")
+    parser.add_argument(
+        "--profile", action="store_true", help="Display profiling information for mask creation"
+    )
+    parser.add_argument(
+        "-l",
+        "--log-level",
+        help="Set the logging level",
+        choices=["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"],
+        default="WARNING",
+    )
+
+    return parser.parse_args()
+
+
+async def process_file(filename, args):
+    # Output the filename to the terminal
+    print(f"Processing file: {filename}")
+
+    config = load_config(args.config)
+    profiling_info = {}
+
+    initialize(args, config, filename)
+
+    # Process the raw data. Convert to EchoData object
+    echodata, encode_mode = read_file(profiling_info=profiling_info, config=config)
+
+    # Compute Sv with encode_mode and save to zarr
+    sv_dataset = compute_sv_with_encode_mode(
+        echodata, encode_mode=encode_mode, profiling_info=profiling_info, config=config
+    )
+    write_processed(sv_dataset, config["output_folder"], config["raw_path"].stem, "zarr")
+    zarr_file = config["output_folder"] + "/" + config["raw_path"].stem + ".zarr"
+    print(f"Computed Sv with encode_mode={encode_mode} and wrote zarr file to: {zarr_file}")
+
+    # Enrich the Sv by adding depth, location, and split-beam angle information
+    sv_enriched = create_enriched_sv(echodata, encode_mode, sv_dataset)
+
```

Comment:
Do we want to be able to run the cli on ek80 data? 
if we do the recommendation would be to add downsampling at this point after enrichement of dataset. 
The downsampling functionality is in the echodata folder in the sv_interpolation module

Reply:
sure, will update